### PR TITLE
Fix build on GNU/Hurd

### PR DESCRIPTION
**Pull request by sthibaul**

### CODE REVIEW

File: apps/transmitmedia.cpp [hunk 1]
```diff
@@ -1005,13 +1005,17 @@ class UdpCommon
 
             if (attr.count("source"))
             {
+#ifdef IP_ADD_SOURCE_MEMBERSHIP
```
Comment (ethouris) [inline apps/transmitmedia.cpp]: Please configure with `--enable-testing` and recompile (`-DENABLE_TESTING=1` if you use cmake directly). Likely the same fix is needed in `testing/testmedia.cpp` file.

Comment (sthibaul) [reply apps/transmitmedia.cpp]: > Likely the same fix is needed in `testing/testmedia.cpp` file.

? No, that file doesn't use `IP_ADD_SOURCE_MEMBERSHIP`.

Comment (ethouris) [reply apps/transmitmedia.cpp]: Oh... you're right. I was stating they are perfectly synchronized. Something to do in the future.

**Comment by ethouris** [comment]
There's still something wrong with the build on Linux and Mingw. Never saw that b4, but it looks as if the `-std=c++11` option wasn't passed correctly. Please update the branch to the latest master - the `VERBOSE=1` option was lately added, at least this could determine the root cause.

File: CMakeLists.txt [hunk 2]
```diff
@@ -28,6 +28,7 @@ set_if(LINUX       ${CMAKE_SYSTEM_NAME} MATCHES "Linux")
 set_if(BSD         ${SYSNAME_LC} MATCHES "bsd$")
 set_if(MICROSOFT   WIN32 AND (NOT MINGW AND NOT CYGWIN))
 set_if(SYMLINKABLE LINUX OR DARWIN OR BSD OR CYGWIN)
+set_if(GNU         ${CMAKE_SYSTEM_NAME} MATCHES "GNU")
```
Comment (ethouris) [inline CMakeLists.txt]: Sorry, overlooked previously. The above `SYMLINKABLE` symbol should be defined for GNU as well. So, move this line before the definition for `SYMLINKABLE` and add "OR GNU" at the end.

Comment (sthibaul) [reply CMakeLists.txt]: Right, done so

**Comment by sthibaul** [comment]
> Please update the branch to the latest master

I had already done so.

>  - the `VERBOSE=1` option was lately added, at least this could determine the root cause.

I don't see that in the master git log.

File: CMakeLists.txt [hunk 3]
```diff
@@ -27,7 +27,8 @@ set_if(DARWIN      ${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
 set_if(LINUX       ${CMAKE_SYSTEM_NAME} MATCHES "Linux")
 set_if(BSD         ${SYSNAME_LC} MATCHES "bsd$")
 set_if(MICROSOFT   WIN32 AND (NOT MINGW AND NOT CYGWIN))
-set_if(SYMLINKABLE LINUX OR DARWIN OR BSD OR CYGWIN)
+set_if(GNU         ${CMAKE_SYSTEM_NAME} MATCHES "GNU")
+set_if(SYMLINKABLE LINUX OR DARWIN OR BSD OR CYGWIN or GNU)
```
Comment (ethouris) [inline CMakeLists.txt]: `OR` uppercase; your version causes a build break

Comment (sthibaul) [reply CMakeLists.txt]: Oh, right, sorry!

**Comment by ethouris** [comment]
We have some weird problem - I confirmed that old compilers and cmake have some problems. Another PR will solve it.

**Comment by ethouris** [comment]
Please rebase this to the latest master. The build break on Travis should disappear.

**Comment by sthibaul** [comment]
That seems to have helped indeed :)

**Comment by ethouris** [comment]
Yeah, the comparison in cmake should not use `${VARIABLE}` expressions (unless you are sure that it resolves to a number). If this results in a string that resolves to a valid variable name, this variable's value will be taken as a value there. This one was working until you introduced a new variable named `GNU`, which accidentally was a value returned by `CMAKE_COMPILER_ID`.

**Pull request merged**